src/java.base/share/classes/jdk/internal/access/JavaLangAccess.java

```diff
@@ -319,13 +322,14 @@ public interface JavaLangAccess {
     String newStringNoRepl(byte[] bytes, Charset cs) throws CharacterCodingException;
 
     /**
-     * Encode the given string into a sequence of bytes using the specified Charset.
-     *
-     * This method avoids copying the String's internal representation if the input
-     * is ASCII.
-     *
-     * This method throws CharacterCodingException instead of replacing when
-     * malformed input or unmappable characters are encountered.
+     * Encode the given string into a sequence of bytes using the specified
+     * {@linkplain java.nio.charset.Charset charset}.
+     * <p>
+     * <b>WARNING: This method return the {@code byte[]} backing the provided
+     * {@code String}, if the input is ASCII.</b>
+     * <p>
+     * This method throws {@code CharacterCodingException} instead of replacing
+     * when malformed input or unmappable characters are encountered.
```

Comment:
Do we have a JBS issue to track renaming the "no replacement" methods? The naming has confused several people as not everyone is familiar with coding error actions.  Here, "no replacement" actually means that coding errors are reported.

Reply:
Created [JDK-8356439](https://bugs.openjdk.org/browse/JDK-8356439) for this purpose.